### PR TITLE
handle page custom buttons for wizard

### LINKED ISSUE #16687: Custom buttons for the wizard page are not displayed
- Version: 1.32.0-insider (user setup)
- Commit: 309e558727225bde9e577f8f566e68f8b657ae45
- OS: Windows_NT x64 10.0.19043

Steps to Reproduce:

1. Modify `/samples/sqlservice/src/controller/mainController.ts` on line 527, replace `wizard.customButtons = ...` with `page1.customButtons = ...`, which essentially registers custom buttons for a certain page, instead of the entire wizard
2. Build the `sqlservice` sample extension and install it in ADS
3. In ADS run the `sqlservice.openWizard` command from command palette
4. Look at the buttons panel at the bottom

Actual result:
image

Expected result:
image

### PR DESCRIPTION
**Pull request by alanrenmsft**
This PR fixes #16687

this features was documented but never actually implemented.

### CODE REVIEW

File: src/sql/workbench/services/connection/browser/connectionDialogWidget.ts [hunk 1]
```diff
@@ -318,10 +318,12 @@ export class ConnectionDialogWidget extends Modal {
 		const divContainer = DOM.append(recentConnectionContainer, DOM.$('.server-explorer-viewlet'));
 		const treeContainer = DOM.append(divContainer, DOM.$('.explorer-servers'));
 		const leftClick = (element: any, eventish: ICancelableEvent, origin: string) => {
-			// element will be a server group if the tree is clicked rather than a item
-			const isDoubleClick = origin === 'mouse' && (eventish as MouseEvent).detail === 2;
-			this._connectionSource = 'recent';
-			this.onConnectionClick(element, isDoubleClick);
+			// element will be a server group if the blank area of the tree is clicked, we should only proceed is a connection profile is selected.
+			if (element instanceof ConnectionProfile) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/connection/browser/connectionDialogWidget.ts:322]: Looks like you have some other changes in here - merge in latest from main?

File: src/sql/workbench/services/dialog/browser/wizardModal.ts [hunk 2]
```diff
@@ -199,6 +199,16 @@ export class WizardModal extends Modal {
 		if (dialogPaneToShow && readHeader) {
 			status(`${dialogPaneToShow.pageNumberDisplayText} ${dialogPaneToShow.title}`);
 		}
+
+		// Remove the current page's custom buttons
+		this._wizard.pages[this._wizard.currentPage]?.customButtons.forEach(button => {
+			this.removeFooterButton(button.label);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/dialog/browser/wizardModal.ts:205]: This looks like it's disposing the button - did you check to make sure that this isn't going to break anything (such as the onClick handlers)?

File: src/sql/workbench/services/connection/browser/connectionDialogWidget.ts (hunk 1 above)
Comment (alanrenmsft) [reply src/sql/workbench/services/connection/browser/connectionDialogWidget.ts:322]: oops, sorry, I should have checked the changes before submitting the PR, I created this branch off the previous PR's branch..., I'll fix it.

**Review by abist — APPROVED** [review]

File: src/sql/workbench/services/dialog/browser/wizardModal.ts (hunk 2 above)
Comment (alanrenmsft) [reply src/sql/workbench/services/dialog/browser/wizardModal.ts:205]: yes, I've tested it, everything works fine

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**